spot-vaults/contracts/charm/UsdcSpotManager.sol

```diff
@@ -87,23 +113,50 @@ contract UsdcSpotManager is Ownable {
         return r;
     }
 
+    /// @notice Updates the active zone definition.
+    function updateActiveZone(Range memory activeZoneDeviation_) external onlyOwner {
+        activeZoneDeviation = activeZoneDeviation_;
+    }
+
+    /// @notice Updates the width of the concentrated liquidity band.
+    function updateConcentratedBand(uint256 concBandDeviationWidth_) external onlyOwner {
+        concBandDeviationWidth = concBandDeviationWidth_;
+    }
+
+    /// @notice Updates the absolute and percentage maximum amount of liquidity
+    ///         in the full range liquidity band.
+    function updateFullRangeLiquidity(
+        uint256 fullRangeMaxUsdcBal_,
+        uint256 fullRangeMaxPerc_
+    ) external onlyOwner {
+        // solhint-disable-next-line custom-errors
+        require(fullRangeMaxPerc_ <= ONE, "InvalidPerc");
+        fullRangeMaxUsdcBal = fullRangeMaxUsdcBal_;
+        fullRangeMaxPerc = fullRangeMaxPerc_;
+    }
+
     //--------------------------------------------------------------------------
     // External write methods
 
     /// @notice Executes vault rebalance.
     function rebalance() public {
         (uint256 deviation, bool deviationValid) = oracle.spotPriceDeviation();
+        bool withinActiveZone = (deviationValid && activeZone(deviation));
+        bool withinActiveZoneBefore = activeZone(prevDeviation);
```

Comment:
Consider caching the boolean `withinActiveZoneBefore`, rather than recomputing it from the cached prevDeviation each time.

Reply:
done